ocs_ci/ocs/pgsql.py

```diff
@@ -0,0 +1,293 @@
+            CommandFailed: If PostgreSQL server setup fails
+
+        """
+        log.info("Deploying postgres database")
+        try:
+            pgsql_service = templating.load_yaml(
+                constants.PGSQL_SERVICE_YAML
+            )
+            pgsql_cmap = templating.load_yaml(
+                constants.PGSQL_CONFIGMAP_YAML
+            )
+            pgsql_sset = templating.load_yaml(
+                constants.PGSQL_STATEFULSET_YAML
+            )
+            pgsql_sset['spec']['replicas'] = replicas
+            self.pgsql_service = OCS(**pgsql_service)
+            self.pgsql_service.create()
+            self.pgsql_cmap = OCS(**pgsql_cmap)
+            self.pgsql_cmap.create()
+            self.pgsql_sset = OCS(**pgsql_sset)
+            self.pgsql_sset.create()
+            self.pod_obj.wait_for_resource(
+                condition='Running',
+                selector='app=postgres',
+                resource_count=replicas,
+                timeout=3600
+            )
+        except (CommandFailed, CalledProcessError) as cf:
+            log.error('Failed during setup of PostgreSQL server')
+            raise cf
+        self.pgsql_is_setup = True
+        log.info("Successfully deploying postgres database")
+
+    def create_pgbench_benchmark(
+        self, replicas, clients=None, threads=None,
+        transactions=None, scaling_factor=None
+    ):
+        """
+        Create pgbench benchmark pods
+
+        Args:
+            replicas (int): Number of pgbench pods to be deployed
+            clients (int): Number of clients
+            threads (int): Number of threads
+            transactions (int): Number of transactions
+            scaling_factor (int): scaling factor
+
+        """
+        for i in range(replicas):
+            log.info("Create resource file for pgbench workload")
+            pg_data = templating.load_yaml(constants.PGSQL_BENCHMARK_YAML)
+            pg_data['metadata']['name'] = 'pgbench-benchmark' + f"{i}"
+            pg_data['spec']['workload']['args']['databases'][0][
+                'host'
+            ] = "postgres-" + f"{i}" + ".postgres"
+
+            if clients is not None:
+                pg_data['spec']['workload']['args']['clients'][0] = clients
+            if threads is not None:
+                pg_data['spec']['workload']['args']['threads'] = threads
+            if transactions is not None:
+                pg_data[
+                    'spec'
+                ]['workload']['args']['transactions'] = transactions
+            if scaling_factor is not None:
+                pg_data[
+                    'spec'
+                ]['workload']['args']['scaling_factor'] = scaling_factor
+            pg_obj = OCS(**pg_data)
+            pg_obj.create()
+
+        # Confirm that expected pgbench pods are spinned
+        log.info("Checking if Getting pgbench pods name")
+        for pgbench_pods in TimeoutSampler(
+            3600, replicas, get_pod_name_by_pattern,
```

Comment:
The wait time depends on the number of pgbench pods to spin. If the replica count is high then the wait time for the pgbench pods will be more.
Please note that the wait time is not for a single pod, it will wait till it matches the total replica count number of pods.
I can set the default timeout as 300sec and pass timeout as arg to this function, so that it can be modified according to the test case.

Reply:
Implemented this change in the updated patch.

Reply:
Sounds good. A note in the docstring about how long it would take based on the replica count will be helpful. i.e., if 1 pod takes 300 seconds, will 10 pods take 3000 seconds or lesser value. I think it is not linear.

Reply:
Ack. Will take this in next commit/PR.